src/main/kotlin/com/swisscom/health/des/cdr/clientvm/config/CdrClientConfig.kt

```diff
@@ -52,33 +60,65 @@ data class CdrClientConfig(
     @PostConstruct
     fun checkAndReport() {
         if (customer.isEmpty()) {
-            error("There where no customer entries configured")
+            error("There were no customer entries configured")
         }
+        sourceTargetFolderOverlap()
         // we don't check target folder for duplicate as this can be configured deliberately by customers
-        if (duplicateFolderUsage() || duplicateUsage(Connector::sourceFolder)) {
-            error("Duplicate folder usage detected. Please make sure that each customer has a unique source and that no target folder is used " +
-                    "at the same time as source folder.")
-        }
+        duplicateSourceFolders()
         checkNoConnectorIdHasTheSameModeDefinedTwice()
-        File(localFolder).mkdirs()
+        allFoldersAreReadWriteable()
+
+        if (localFolder.exists() && !localFolder.isDirectory()) {
+            error("Local folder is not a directory: '$localFolder'")
+        }
+
+        if (!localFolder.exists()) {
```

Comment:
This check doesn't make sense if we throw an exception above if the localFolder doesn't exist

Reply:
The check above is for the existence of a file with the path that is supposed to be a directory. Only in that case do we error out.